packages/actionmenu/src/react/index.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { HTMLAttributes } from 'react'
+import { css, compose, keyframes } from 'glamor'
+import { useMenuRef } from '@pluralsight/ps-design-system-util'
+import filterReactProps from '@pluralsight/ps-design-system-filter-react-props'
+import {
+  handleMenuKeyDownEvents,
+  handleMenuKeyUpEvents,
+  useCloseOnDocumentEvents,
+  RefForwardingComponent,
+} from '@pluralsight/ps-design-system-util'
+import { ActionMenuContext } from './context'
+import { Divider } from './divider'
+import { Item } from './item'
+import { Ellipsis } from './ellipsis'
+import { Icon } from './icon'
+import stylesheet from '../css/index'
+import * as vars from '../vars/index'
+
+const slide = keyframes(
+  stylesheet['@keyframes psds-actionmenu__keyframes__slide']
+)
+
+const styles = ({ origin }) =>
+  compose(
+    css(stylesheet['.psds-actionmenu']),
+    css(stylesheet[`.psds-actionmenu--origin-${origin}`]),
+    css(stylesheet['.psds-actionmenu__animation']({ slide }))
+  )
+
+interface ActionMenuStatics {
+  Divider: typeof Divider
+  Item: typeof Item
+  Ellipsis: typeof Ellipsis
+  Icon: typeof Icon
+  origins: typeof vars.origins
+  tagName: typeof vars.tagName
+  useMenuRef: typeof useMenuRef
+}
+
+interface Props extends HTMLAttributes<HTMLUListElement> {
+  children: React.ReactNode | Array<React.ReactNode>;
+  onClose: () => void;
+  origin: string;
```

Comment:
we can provide more helpful information to consumers in these types by limiting the strings to only the available vars

```suggestion
  origin: ValueOf<typeof string>;
```

`ValueOf` is in the utils package